Suggestions and Activity Pub spec (#1)
foranetworks/protocol: core/core.md

**User**: This is a member of the Fora community, they may or may not host the Fora community. A member that hosts the Fora community will specifically be termed **Host** and a member isn't a host and relies on a Host to connect to the community will be specifically termed **Light User**. (NOTE: This is analogous to the Full Node/Light Client distinction in blockchains). Users have a unique name in the community and an associated public key.

> **Host**: As mentioned above, a host is a member that maintains the full state of the community. They will receive messages from other users of the community over a gossip network and update their own state, as well as gossip any updates to the rest of the network. They may or may not serve as an access point for Light users. Hosts will listen to events on the blockchain in order to keep track of any decisions made there. An honest Host will delete any banned content from their local copy of the community and ignore any banned user by refusing to accept and propogate their messages.

Reviewer: I get that we may want to separate out the ActivityPub specifics from the core document. But Hosts must listen to blockchain decisions as well as community interactions. Maybe rephrase to:

```suggestion
**Host**: As mentioned above, a host is a member that maintains the full state of the community. They will receive messages from other users of the community over a gossip network and update their own state, as well as gossip any updates to the rest of the network. They may or may not serve as an access point for Light users. Hosts will record and propogate community interactions, and they must also listen to events on the blockchain in order to abide by any decisions made there. An honest Host will delete any banned content from their local copy of the community and ignore any banned user by refusing to accept and propogate their messages.
```

As mentioned above, for an honest users to successfully connect to the rest of the honest community; they must not be eclipsed by dishonest Hosts. If an honest user has no choice but to access the community through a dishonest Host, then Fora can make no guarantee that the user can be protected from banned users and banned content.

> Thus the Fora protocol assumes that the peer-to-peer network is sufficiently distributed and decentralized such that no user can be fully eclipsed by dishonest Hosts.

Reviewer: ```suggestion
Thus the Fora protocol assumes that the peer-to-peer network is sufficiently distributed and decentralized such that no member can be fully eclipsed by dishonest Hosts.
```
